### PR TITLE
fix(logging): use DD_TRACE_LOG_FILE_SIZE_BYTES when specifying log file size bytes

### PR DESCRIPTION
**Pull request by wantsui**
## Description

Use `DD_TRACE_LOG_FILE_SIZE_BYTES` instead of `DD_TRACE_FILE_SIZE_BYTES` when specifying log file size bytes.

## Checklist
- [x] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [x] Ensure tests are passing for affected code.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Testing strategy

The only testing changes is on the `test_tracer_logging.py` file.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] PR cannot be broken up into smaller PRs.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

**Review by brettlangdon — DISMISSED** [review]
lgtm, curious if others think we need to continue to support the "old" env name or a quick release of `1.3.1` is enough to address this?

**Comment by majorgreys** [comment]
@Mergifyio backport 1.3

File: releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Fixed the environment variable used for log file size byes by replacing ``DD_TRACE_FILE_SIZE_BYTES`` with ``DD_TRACE_LOG_FILE_SIZE_BYTES``.
```
Comment (majorgreys) [inline releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml:4]:
```suggestion
    Fixed the environment variable used for log file size bytes to be ``DD_TRACE_LOG_FILE_SIZE_BYTES`` as documented.
```

**Review by mabdinur — COMMENTED** [review]
LGTM after the release note is reformatted

**Review by majorgreys — DISMISSED** [review]

File: releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml:1]: is this needed? we haven't released 1.3 yet so messaging should be required

File: releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+fixes:
+  - |
+    Fixed the environment variable used for log file size bytes to be ``DD_TRACE_LOG_FILE_SIZE_BYTES`` as documented.
```
Comment (mabdinur) [inline releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml:4]:
```suggestion
    Fixed the environment variable used for log file size bytes to be ``DD_TRACE_LOG_FILE_SIZE_BYTES`` as documented.

```

**Review by mabdinur — APPROVED** [review]

**Comment by ZStriker19** [comment]
I think with the release note we should be ok to not support the old one, since it never worked anyways.

**Review by ZStriker19 — APPROVED** [review]

File: releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml (hunk 2 above)
Comment (brettlangdon) [reply releasenotes/notes/fix-ddtrace-file-logs-bytes-env-var-ed690ffe3fb96f7e.yaml:1]: 1.3 has now been released

**Comment by brettlangdon** [comment]
@Mergifyio backport 1.3

**Pull request merged**